src/main/java/com/matchbook/sdk/configs/ClientConnectionManager.java

```diff
@@ -0,0 +1,52 @@
+package com.matchbook.sdk.configs;
+
+import java.util.concurrent.TimeUnit;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.matchbook.sdk.ClientConfig;
+import com.squareup.okhttp.OkHttpClient;
+
+public final class ClientConnectionManager {
+
+    private final ClientConfig clientConfig;
+    private final OkHttpClient httpClient;
+    private final ObjectMapper mapper;
+
+    public ClientConnectionManager(ClientConfig clientConfig) {
+        this.clientConfig = clientConfig;
+        this.httpClient = buildOkHttpClient();
+        this.mapper = buildObjectMapper();
+    }
+
+    private ObjectMapper buildObjectMapper() {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+        mapper.setPropertyNamingStrategy(PropertyNamingStrategy.KEBAB_CASE);
+        mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        return mapper;
+    }
+
+    private OkHttpClient buildOkHttpClient() {
+        OkHttpClient okHttpClient = new OkHttpClient();
+        okHttpClient.setConnectTimeout(5, TimeUnit.SECONDS);
+        okHttpClient.setWriteTimeout(5, TimeUnit.SECONDS);
+        okHttpClient.setReadTimeout(5, TimeUnit.SECONDS);
+        okHttpClient.setFollowRedirects(false);
+        return okHttpClient;
+    }
+
+    public ClientConfig getClientConfig() {
+        return clientConfig;
+    }
+
+    public OkHttpClient getHttpClient() {
```

Comment:
Might exist cases where the HTTP client has to be re-created?

Reply:
I don't think so, you will update http client in sdk usage, do you have an example when we might need it?

Reply:
Not really. I remember from my Android experience it was a good idea to recreate the `OkHttpClient` in case the network context changed, such as switching from WiFi to mobile connection. Though I can't imagine a situation in which this could be needed in an SDK, this why I'm asking you.

Reply:
yeah, i don't think it might change in the middle of the work, so we can leave it like that for now